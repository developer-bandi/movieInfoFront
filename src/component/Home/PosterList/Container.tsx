import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PosterList from './Presentational';
import imagePreload from '../../../lib/imagePreload';
import { ReducerType } from '../../../store';
import { movePage } from '../../../store/homePoster/Reducer';

const NowShowingContainer = ({ title }: { title: string }) => {
  const dispatch = useDispatch();
  const postersData = useSelector((state: ReducerType) => state.homePoster);
  const [nowPage, setNowPage] = useState(
    title === '현재상영중'
      ? postersData.content.page.nowShowingInfo
      : postersData.content.page.nowCommingInfo
  );

  useEffect(() => {
    if (postersData.content.posterList !== undefined) {
      postersData.content.posterList.nowCommingInfo
        .concat(postersData.content.posterList.nowShowingInfo)
        .forEach((posterData) => {
          imagePreload(posterData.posterPath);
        });
    }
  }, [postersData]);
  //이미지를 미리 로드하여 포스터 전환시 포스터를 빠르게 보여줍니다.

  const changePage = (position: string) => {
    setNowPage(position === 'prev' ? nowPage - 1 : nowPage + 1);
    dispatch(
      movePage({
        kind: title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo',
        page: nowPage + 1,
      })
    );
  };

  return (
    <>
      <PosterList
        postersData={postersData}
        title={title}
        changePage={changePage}
        page={
          postersData.content.page[
            title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
          ]
        }
      />
    </>
  );
};

export default NowShowingContainer;
