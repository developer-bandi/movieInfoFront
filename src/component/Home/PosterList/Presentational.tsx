import { FiChevronLeft, FiChevronRight } from 'react-icons/fi';
import nullmovie from '../../../imgs/nullmovie.webp';
import styles from './Style';
import { HomePosterState } from '../../../store/homePoster/Reducer';
import React from 'react';

interface PosterListProps {
  title: string;
  postersData: HomePosterState;
  changePage: (position: string) => void;
  page: number;
}
const PosterListPresentational = ({
  title,
  postersData,
  changePage,
  page,
}: PosterListProps) => {
  if (postersData.loading) {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          <styles.NullButton />
          {new Array(5).fill(0).map((zero, index) => {
            return (
              <styles.NullPosterBlock key={index}></styles.NullPosterBlock>
            );
          })}
          <styles.NullButton />
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  } else if (postersData.error) {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          <styles.Error>에러 발생</styles.Error>
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  } else {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          {page === 1 ? (
            <styles.NullButton />
          ) : (
            <styles.PageButton
              onClick={() => changePage('prev')}
              key={`prev${title}`}
            >
              <FiChevronLeft size={40} />
            </styles.PageButton>
          )}
          {postersData.content.posterList !== undefined &&
            postersData.content.posterList[
              title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
            ]
              .slice((page - 1) * 5, 5 * page)
              .map((posterData, index) => {
                return (
                  <React.Fragment key={posterData.id}>
                    <styles.PosterBlock to={`/moviedetail/${posterData.id}`}>
                      {posterData.posterPath === null ? (
                        <styles.MovieImg src={nullmovie} alt="notPoster" />
                      ) : (
                        <styles.MovieImg
                          src={`https://image.tmdb.org/t/p/w500${posterData.posterPath}`}
                          alt="poster"
                        ></styles.MovieImg>
                      )}
                      <styles.MovieInfo className="movieinfo">
                        {posterData.overview.length > 170
                          ? posterData.overview.substring(0, 170) + '...'
                          : posterData.overview}
                      </styles.MovieInfo>
                      <styles.MovieRating className="rating">
                        평점 {posterData.voteAverage}
                      </styles.MovieRating>
                      <styles.MoiveName className="moviename">
                        {posterData.title.length > 12
                          ? posterData.title.substring(0, 12) + '...'
                          : posterData.title}
                      </styles.MoiveName>
                    </styles.PosterBlock>
                  </React.Fragment>
                );
              })}
          {
            postersData.content.posterList === undefined ||
            page * 5 >=
              postersData.content.posterList[
                title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
              ].length ? (
              <styles.NullButton />
            ) : (
              <styles.PageButton
                onClick={() => changePage('next')}
                key={`next${title}`}
              >
                <FiChevronRight size={40} />
              </styles.PageButton>
            ) /*이후로 넘길수 없으면 빈박스만 렌더링하고 버튼을 누르면 start와 end값에 5를 추가 */
          }
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  }
};

//이미지를 미리 로드해 두는 함수이다.

export default PosterListPresentational;
