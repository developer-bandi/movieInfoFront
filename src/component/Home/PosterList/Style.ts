import styled from 'styled-components';
import { Link } from 'react-router-dom';

const MainBlock = styled.section`
  width: 89rem;
  height: 28rem;
  margin: auto;
  margin-top: 3rem;
  margin-bottom: 7rem;

  @media screen and (max-width: 500px) {
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    width: 100%;
    height: 11rem;
  }
`;

const TitleBlock = styled.div`
  display: flex;
  width: 89rem;
  @media screen and (max-width: 500px) {
    width: 100%;
  }
`;

const Title = styled.h2`
  width: 89rem;
  padding-left: 3rem;
  margin-bottom: 1rem;
  font-weight: bold;
  font-size: 50px;

  @media screen and (max-width: 500px) {
    width: 100%;
    margin-bottom: 0;
    padding: 0;
    font-size: 15px;
    margin-left: 1rem;
  }
`;

const PosterListBlock = styled.div`
  width: 89rem;
  height: 20rem;
  display: flex;
  align-items: center;

  @media screen and (max-width: 500px) {
    width: 100%;
    height: 10rem;
  }
`;

const Error = styled.div`
  font-size: 40px;
  color: gray;
  margin: auto;
`;

const PosterBlock = styled(Link)`
  position: relative;
  top: 1rem;
  border-radius: 10px;
  color: inherit;
  text-decoration: none;
  @media (hover: hover) {
    &:hover {
      & > img {
        filter: brightness(20%);
      }
      & > .movieinfo {
        visibility: visible;
        color: white;
      }
    }
  }

  & + & {
    margin-left: 2rem;
  }
  @media screen and (max-width: 500px) {
    width: 4rem;
    height: 8rem;
    top: 0;
    & + & {
      margin-left: 0.3rem;
    }
  }
`;

const NullPosterBlock = styled.div`
  position: relative;
  top: 1rem;
  border-radius: 10px;
  color: inherit;
  text-decoration: none;
  & + & {
    margin-left: 2rem;
  }

  @media screen and (max-width: 500px) {
    width: 4rem;
    height: 8rem;
    top: 0;
    & + & {
      margin-left: 0.3rem;
    }
  }
`;

const MovieImg = styled.img`
  border-radius: inherit;
  width: 15rem;
  height: 20rem;
  @media screen and (max-width: 500px) {
    width: 4rem;
    height: 7rem;
  }
`;

const MovieInfo = styled.div`
  visibility: hidden;
  position: absolute;
  top: 0;
  left: 0;
  padding: 1rem;
  line-height: 1.5;
  @media screen and (max-width: 500px) {
    display: none;
  }
`;

const MovieRating = styled.div`
  position: absolute;
  font-size: 20px;
  font-weight: bold;
  top: 17rem;
  color: white;
  width: 100%;
  height: 3rem;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0 0 10px 10px;
  @media screen and (max-width: 500px) {
    display:none;
`;

const MoiveName = styled.div`
  text-align: center;
  font-size: 20px;
  font-weight: 600;
  @media screen and (max-width: 500px) {
    font-size: 10px;
  }
`;

const PageButton = styled.div`
  width: 3rem;
  height: 3rem;
  &:hover {
    color: gray;
  }
  @media screen and (max-width: 500px) {
    position: relative;
    width: 1rem;
    height: 1rem;
    bottom: 0.5rem;
  }
`;

const NullButton = styled.div`
  width: 3rem;
  height: 3rem;
  @media screen and (max-width: 500px) {
    width: 1rem;
    height: 1rem;
  }
`;

const styles = {
  MainBlock,
  MoiveName,
  MovieImg,
  MovieInfo,
  MovieRating,
  NullButton,
  NullPosterBlock,
  PageButton,
  PosterBlock,
  PosterListBlock,
  Error,
  Title,
  TitleBlock,
};

export default styles;
