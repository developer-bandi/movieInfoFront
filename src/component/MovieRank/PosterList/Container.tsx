import { useSelector } from 'react-redux';
import PosterListPresentational from './Presentational';
import { ReducerType } from '../../../store';

const PosterListContainer = () => {
  const movieRankList = useSelector((state: ReducerType) => state.movieRank);

  return <PosterListPresentational movieRankList={movieRankList} />;
};

export default PosterListContainer;
