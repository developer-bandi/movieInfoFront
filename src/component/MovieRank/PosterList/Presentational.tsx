import styles from './Style';
import nullMovie from '../../../imgs/nullmovie.webp';
import OrderButtonContainer from '../OrderButton/Container';
import Except from '../../common/Except/Presentational';
import { MovieRankState } from '../../../store/movieRank/Reducer';

interface PosterListProps {
  movieRankList: MovieRankState;
}

const PosterListPresentational = ({ movieRankList }: PosterListProps) => {
  if (movieRankList.loading) {
    return (
      <styles.MainBlock>
        <OrderButtonContainer />
        <styles.PosterListBlock>
          {new Array(20).fill(0).map((zero, index) => {
            return <styles.LoadingPoster key={index} active={index} />;
          })}
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  } else if (movieRankList.error) {
    return (
      <styles.MainBlock>
        <Except text="에러가 발생하였습니다" />
      </styles.MainBlock>
    );
  } else {
    return (
      <styles.MainBlock>
        <OrderButtonContainer />
        <styles.PosterListBlock>
          {movieRankList.content.movieInfo !== undefined &&
            movieRankList.content.movieInfo[movieRankList.content.type].map(
              (movieInfo, index) => {
                return (
                  <styles.PosterBlock
                    key={index}
                    to={`/moviedetail/${movieInfo.id}`}
                    active={index}
                  >
                    <styles.MovieImg
                      src={
                        movieInfo.posterPath === null
                          ? nullMovie
                          : `https://image.tmdb.org/t/p/w500${movieInfo.posterPath}`
                      }
                      alt={
                        movieInfo.posterPath === null
                          ? 'loading'
                          : 'moviePoster'
                      }
                    />
                    <styles.MovieNum className="num">
                      {index + 1}
                    </styles.MovieNum>
                    <styles.MovieInfo className="movieinfo">
                      {movieInfo.overview.length > 220
                        ? movieInfo.overview.substring(0, 220) + '...'
                        : movieInfo.overview}
                    </styles.MovieInfo>
                    <styles.MovieRating className="rating">
                      평점 {movieInfo.voteAverage}
                    </styles.MovieRating>
                    <styles.MoiveName className="moviename">
                      {movieInfo.title.length > 12
                        ? movieInfo.title.substring(0, 12) + '...'
                        : movieInfo.title}
                    </styles.MoiveName>
                  </styles.PosterBlock>
                );
              }
            )}
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  }
};

export default PosterListPresentational;
