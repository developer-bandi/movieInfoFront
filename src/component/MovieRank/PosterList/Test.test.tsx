import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import PosterList from './Presentational';
import configureMockStore from 'redux-mock-store';

describe('PosterList Presentational', () => {
  const topRatedMovieInfo = {
    id: 1,
    title: 'topRated',
    overview: 'topRatedMovieInfotestOverview',
    voteAverage: 9.0,
    posterPath: 'topRatedMovieInfotestPoserPath',
  };
  const popularMovieInfo = {
    id: 2,
    title: 'popular',
    overview: 'populartestOverview',
    voteAverage: 8.0,
    posterPath: 'populartestPoserPath',
  };
  const movieRankList = {
    content: {
      type: 'popular',
      movieInfo: {
        topRated: [topRatedMovieInfo],
        popular: [popularMovieInfo],
      },
    },
    loading: false,
    error: false,
  };
  const mockStore = configureMockStore();
  const store = mockStore({
    movieRank: {
      content: { type: 'popular' },
    },
  });
  const wrapper = ({ children }: { children: React.ReactElement }) => (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );
  it('로딩중', () => {
    const movieRankList = {
      content: { type: 'popular' },
      loading: true,
      error: false,
    };
    const utils = render(<PosterList movieRankList={movieRankList} />, {
      wrapper,
    });
    expect(utils.container).toMatchSnapshot();
  });
  it('에러 발생', () => {
    const movieRankList = {
      content: { type: 'popular' },
      loading: false,
      error: true,
    };
    const utils = render(<PosterList movieRankList={movieRankList} />, {
      wrapper,
    });
    expect(utils.container).toMatchSnapshot();
    screen.getByText('에러가 발생하였습니다');
  });
  it('popular 출력', () => {
    const utils = render(<PosterList movieRankList={movieRankList} />, {
      wrapper,
    });
    expect(utils.container).toMatchSnapshot();
    screen.getByText('popular');
    screen.getByText('populartestOverview');
  });
  it('topRated 출력', () => {
    movieRankList.content.type = 'topRated';
    const utils = render(<PosterList movieRankList={movieRankList} />, {
      wrapper,
    });
    expect(utils.container).toMatchSnapshot();
    screen.getByText('topRated');
    screen.getByText('topRatedMovieInfotestOverview');
  });
});
