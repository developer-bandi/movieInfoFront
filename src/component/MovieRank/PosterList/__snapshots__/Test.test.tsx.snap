// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterList <Presentational> popular 출력 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <a
        active="0"
        class="sc-dkPtRN EVaWh"
        href="/moviedetail/2"
      >
        <img
          alt="moviePoster"
          class="sc-hKwDye ibWAQv"
          src="https://image.tmdb.org/t/p/w500populartestPoserPath"
        />
        <div
          class="sc-eCImPb izzlkg num"
        >
          1
        </div>
        <div
          class="sc-jRQBWg knOWJz movieinfo"
        >
          populartestOverview
        </div>
        <div
          class="sc-gKclnd iMzDwk rating"
        >
          평점 
          8
        </div>
        <div
          class="sc-iCfMLu yxOVc moviename"
        >
          popular
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PosterList <Presentational> topRated 출력 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <a
        active="0"
        class="sc-dkPtRN EVaWh"
        href="/moviedetail/1"
      >
        <img
          alt="moviePoster"
          class="sc-hKwDye ibWAQv"
          src="https://image.tmdb.org/t/p/w500topRatedMovieInfotestPoserPath"
        />
        <div
          class="sc-eCImPb izzlkg num"
        >
          1
        </div>
        <div
          class="sc-jRQBWg knOWJz movieinfo"
        >
          topRatedMovieInfotestOverview
        </div>
        <div
          class="sc-gKclnd iMzDwk rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-iCfMLu yxOVc moviename"
        >
          topRated
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PosterList <Presentational> 로딩중 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <div
        class="sc-furwcr fKsSdm"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
    </div>
  </div>
</div>
`;

exports[`PosterList <Presentational> 에러 발생 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-iqseJM exHIdu"
    >
      <div
        class="sc-crHmcD jRVExl"
        data-testid="except"
      >
        에러가 발생하였습니다
      </div>
    </div>
  </div>
</div>
`;

exports[`PosterList Presentational popular 출력 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <a
        active="0"
        class="sc-dkPtRN EVaWh"
        href="/moviedetail/2"
      >
        <img
          alt="moviePoster"
          class="sc-hKwDye ibWAQv"
          src="https://image.tmdb.org/t/p/w500populartestPoserPath"
        />
        <div
          class="sc-eCImPb izzlkg num"
        >
          1
        </div>
        <div
          class="sc-jRQBWg knOWJz movieinfo"
        >
          populartestOverview
        </div>
        <div
          class="sc-gKclnd iMzDwk rating"
        >
          평점 
          8
        </div>
        <div
          class="sc-iCfMLu yxOVc moviename"
        >
          popular
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PosterList Presentational topRated 출력 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <a
        active="0"
        class="sc-dkPtRN EVaWh"
        href="/moviedetail/1"
      >
        <img
          alt="moviePoster"
          class="sc-hKwDye ibWAQv"
          src="https://image.tmdb.org/t/p/w500topRatedMovieInfotestPoserPath"
        />
        <div
          class="sc-eCImPb izzlkg num"
        >
          1
        </div>
        <div
          class="sc-jRQBWg knOWJz movieinfo"
        >
          topRatedMovieInfotestOverview
        </div>
        <div
          class="sc-gKclnd iMzDwk rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-iCfMLu yxOVc moviename"
        >
          topRated
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PosterList Presentational 로딩중 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-pVTFL iznEeA"
    >
      <div
        class="sc-jrQzAO dnGTOR"
      >
        <div
          class="sc-kDTinF eHahVc"
          data-testid="popularOrderButton"
        >
          인기순
        </div>
        <div
          class="sc-kDTinF cSxYEp"
          data-testid="topRatedOrderButton"
        >
          별점순
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ ddJUoe"
    >
      <div
        class="sc-furwcr fKsSdm"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr gPYeB"
      />
      <div
        class="sc-furwcr bgqqZp"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
      <div
        class="sc-furwcr jtieLe"
      />
    </div>
  </div>
</div>
`;

exports[`PosterList Presentational 에러 발생 1`] = `
<div>
  <div
    class="sc-bdvvtL ckomfZ"
  >
    <div
      class="sc-iqseJM exHIdu"
    >
      <div
        class="sc-crHmcD jRVExl"
        data-testid="except"
      >
        에러가 발생하였습니다
      </div>
    </div>
  </div>
</div>
`;
